pkg/transport/generic_transport_controller.go

```diff
@@ -385,6 +424,22 @@ func (c *genericTransportController) updateWrappedObjectsAndFinalizer(ctx contex
 	return nil
 }
 
+func (c *genericTransportController) valuesForDestination(bindingName string, dest v1alpha1.Destination) map[string]string {
+	c.addBindingCare(bindingName, dest) // ensure any change notification received after this causes reconsideration
+	cluster, err := c.inventoryLister.Get(dest.ClusterId)
+	if err != nil {
+		// This should be a transient condition that is already going to be remedied,
+		// because this means that the caller is working on old information.
+		// Just return anything in this case.
+		c.logger.V(2).Info("Failed to get inventory object", "name", dest.ClusterId, "err", err)
+		return nil
```

Comment:
this assumption is wrong. if you return nil it will not be re-queued for re-evaluation.

Reply:
I am working on a revision that includes expanding this comment to reference the bigger picture, hopefully making it clearer why re-queuing is not needed in this situation.

Reply:
You can see the revision now.

Reply:
On second thought, while it is not incorrect to proceed with an arbitrary map, I think that it would indeed be better to abort the current work (since, as noted, it is already known to be moot).